test/e2e/tests/request-queuing/watchAsset-switchChain-watchAsset.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Suite } from 'mocha';
+import { withFixtures, WINDOW_TITLES } from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+import { Driver } from '../../webdriver/driver';
+import { loginWithBalanceValidation } from '../../page-objects/flows/login.flow';
+import TestDapp from '../../page-objects/pages/test-dapp';
+import TransactionConfirmation from '../../page-objects/pages/confirmations/redesign/transaction-confirmation';
+import AddTokensModal from '../../page-objects/pages/dialog/add-tokens';
+import ReviewPermissionsConfirmation from '../../page-objects/pages/confirmations/redesign/review-permissions-confirmation';
+
+describe('Request Queue WatchAsset -> SwitchChain -> WatchAsset', function (this: Suite) {
+  it('should not batch subsequent watchAsset token into first watchAsset confirmation with a switchChain in the middle', async function () {
+    const port = 8546;
+    const chainId = 1338;
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerDoubleNode()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        localNodeOptions: [
+          {
+            type: 'anvil',
+          },
+          {
+            type: 'anvil',
+            options: {
+              port,
+              chainId,
+            },
+          },
+        ],
+        title: this.test?.fullTitle(),
+      },
+
+      async ({ driver }: { driver: Driver }) => {
+        await loginWithBalanceValidation(driver);
+
+        const testDapp = new TestDapp(driver);
+        await testDapp.openTestDappPage();
+
+        // Create Token
+        await testDapp.clickCreateToken();
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const transactionConfirmation = new TransactionConfirmation(driver);
+        await transactionConfirmation.clickFooterConfirmButton();
+
+        // Wait for token address to populate in dapp
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await testDapp.checkTokenAddressesValue(
+          '0x581c3C1A2A4EBDE2A0Df29B5cf4c116E42945947',
+        );
```

Comment:
oh I just realized another small nit now. We should be able to remove all these steps, if we use instead the injected token like this: `smartContract: SMART_CONTRACTS.HST,` in the withFixtures method and then open the test dapp with the contract address 

```
const contractAddress = await (
    contractRegistry as ContractAddressRegistry
  ).getContractAddress(SMART_CONTRACTS.HST);
await testDapp.openTestDappPage({ contractAddress, url: DAPP_URL });
```



```suggestion
```